python/lsst/sconsUtils/state.py

```diff
@@ -326,6 +326,14 @@ def ClassifyCc(context):
                 env['CXX'] = CXX
         conf = env.Configure(custom_tests={'ClassifyCc': ClassifyCc})
         env.whichCc, env.ccVersion = conf.ClassifyCc()
+
+        # If we have picked up a default compiler called gcc that is really
+        # clang, we call it clang to avoid confusion (gcc on macOS has subtly
+        # different options)
+        if not env['cc'] and env.whichCc == "clang" and env['CC'] == "gcc":
```

Comment:
`env['cc']` is empty, but `env['CC']` is not.... Confusing! (Not your fault!)
Might `env['CC']` be `gcc` with some extra version cruft, e.g., `gcc-4.8.4`?

Reply:
`env['cc']` is the external command line override for the compiler and `env['CC']` is the compiler that we are actually using. If there is extra cruft in the gcc name then it's highly likely that this is not a `clang` -- if it is a `clang` then it makes no sense for it to be a `gcc-n`.